pipelines/deduplication/acteur_deduplicate.py

```diff
@@ -0,0 +1,223 @@
+    print(f"{db_url=}")
+    print(f"{db_engine=}")
+    print(f"{clusters_done_path=}")
+    print(f"{len(clusters_done_list)=}")
+
+    # Doesn't understand that db_engine is of proper type Engine
+    return db_engine, clusters_done_list, clusters_done_path  # type: ignore
+
+
+def main() -> None:
+    """Fonction principale qui regroupe toute là logique de déduplication"""
+    banner(__file__)
+
+    DRY_RUN = Confirm.ask("DRY RUN (càd pas de modifications DB)?")
+
+    print(f"{DRY_RUN=}")
+    if not DRY_RUN:
+        if not Confirm.ask("\n🔴 Modifications DB à venir: continuer? 🔴"):
+            raise SystemExit("Interrompu par l'utilisateur")
+
+    # ------------------------------------------
+    # SETUP D'ENVIRONNEMENT
+    # ------------------------------------------
+    db_engine, clusters_done_list, clusters_done_path = environment_setup()
+
+    # ------------------------------------------
+    # DONNEES D'ENTREE
+    # ------------------------------------------
+    # Chaque ligne est un cluster_id <-> acteur identifiant_unique
+    df_src = source_data_get(CLUSTERING_CSV_FILEPATH)
+
+    # ------------------------------------------
+    # CALCUL DU NOMBRE D'ENFANTS PAR PARENT
+    # ------------------------------------------
+    # Utiliser dans la logique de clustering
+    # pour privilégier les parents avec le plus d'enfants
+    banner("CALCUL DU NOMBRE D'ENFANTS SUR LES PARENTS")
+    # TODO: remplacer par une requête Django utilisant modèle RevisionActeur
+    df_revision_with_parents_db = pd.read_sql(
+        """SELECT identifiant_unique, parent_id
+        FROM qfdmo_revisionacteur WHERE parent_id IS NOT NULL;""",
+        db_engine,
+    )
+    # MAPPING parent_id -> identifiant_unique
+    parent_ids_to_identifiant_ids_db = df_mapping_one_to_many(
+        "parent_ids_to_identifiant_ids_db",
+        df_revision_with_parents_db,
+        "parent_id",
+        "identifiant_unique",
+    )
+    # MAPPING parent_id -> nombre d'enfants
+    parent_ids_to_children_count_db = {
+        k: len(v) for k, v in parent_ids_to_identifiant_ids_db.items()
+    }
+
+    # ------------------------------------------
+    # RECUPERATION IDS DES SOURCES PREFEREES
+    # ------------------------------------------
+    # Pour privilégier certaines sources au moment de la création des parents
+    banner("RECUPERATION IDS DES SOURCES PREFEREES")
+    # TODO: remplacer par une requête Django utilisant modèle Source
+    query_source = "SELECT id, code FROM qfdmo_source"
+    sources = pd.read_sql_query(query_source, db_engine)
+    sources_preferred_ids = sources[sources["code"].isin(SOURCES_PREFERRED_CODES)][
+        "id"
+    ].tolist()
+    print(f"{SOURCES_PREFERRED_CODES=}", f"{sources_preferred_ids=}")
+
+    # ------------------------------------------
+    # GESTION DES CLUSTERS
+    # ------------------------------------------
+    banner("GESTION DES CLUSTERS")
+    df_src = df_src.sort_values("cluster_id", ascending=True)
```

Comment:
Je trouverais ça pratique de nommer chaque étape, pour bien savoir ce qu'on manipule lorsqu'on execute une fonction sur un dataframe. 

```suggestion
    df_sorted_by_cluster_id = df_src.sort_values("cluster_id", ascending=True)
```

Je ne sais pas quelles sont les _best practices_ en la matière, après une brève recherche, il ne semble pas y avoir consensus, en vrac : 
- https://datascience.stackexchange.com/questions/18989/naming-conventions-for-dataframes/19004#19004 
- https://stackoverflow.com/a/60546217
- https://medium.com/@agusmahari/pandas-best-practices-for-writing-clean-code-in-pandas-5b378ba757af

Reply:
partant pour une session nomenclature / coding styles où on s'accorde là dessus